### PR TITLE
Rewrite HttpFaultInjector to use ILogger and built-in ASP.NET Core logging

### PR DESCRIPTION
**Pull request by lmolkova**
- Logging:
  - Leverages built-in ASP.NET Core & HttpClientFactory logging
  - Uses ILogger for a few remaining messages not covered by default logging
  - Adds OTel Console exporter to print correlated and structured logs to console
  - Configures logging to capture headers and other info captured before
- Configuration:
  - Updates default endpoints and allows to leverage default kestrel config via env vars/cmd line args
- Misc refactorings

Example:
```
LogRecord.Timestamp:               2023-12-19T21:46:21.1159635Z
LogRecord.TraceId:                 6728003b59ab450cb730ab8c5670f3f9
LogRecord.SpanId:                  5f82c7d70faf07d6
LogRecord.TraceFlags:              None
LogRecord.CategoryName:            Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware
LogRecord.Severity:                Info
LogRecord.SeverityText:            Information
LogRecord.FormattedMessage:        Request:
Protocol: HTTP/1.1
Method: GET
Scheme: http
PathBase:
Path: /path/to/something
Accept: */*
Host: localhost:7777
User-Agent: curl/8.4.0
X-Upstream-Base-Uri: https://example.com
x-ms-faultinjector-response-option: f
LogRecord.Body:                    Request:
Protocol: HTTP/1.1
Method: GET
Scheme: http
PathBase:
Path: /path/to/something
Accept: */*
Host: localhost:7777
User-Agent: curl/8.4.0
X-Upstream-Base-Uri: https://example.com
x-ms-faultinjector-response-option: f
LogRecord.Attributes (Key:Value):
    Protocol: HTTP/1.1
    Method: GET
    Scheme: http
    PathBase:
    Path: /path/to/something
    Accept: */*
    Host: localhost:7777
    User-Agent: curl/8.4.0
    X-Upstream-Base-Uri: https://example.com
    x-ms-faultinjector-response-option: f
LogRecord.EventId:                 1
LogRecord.EventName:               RequestLog
```

### CODE REVIEW

File: tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "Logging": {
+    "LogLevel": {
+      "Azure": "Information",
+      "Default": "Warning",
+      "System.Net.Http.HttpClient.upstream": "Trace",
+      "Microsoft.AspNetCore.HttpLogging.HttpLoggingMiddleware": "Information"
+    }
+  }
+}
```
Comment (mikeharder) [inline tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/appsettings.json:10]:
```suggestion
}

```

**Review by mikeharder — APPROVED** [review]
Looks great.  Thank you!!!

**Comment by lmolkova** [comment]
@mikeharder I've rebased this change on top of streaming and changed controller to middleware (thanks for the suggestion earlier).

LMK if you want to take another too.
I've done some testing locally and going to run storage tests against this version to check it.

File: tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs [hunk 2]
```diff
@@ -71,364 +30,59 @@ public static void Main(string[] args)
...
-            Console.WriteLine();
-
-            Log("Sending downstream response...");
-
-            response.StatusCode = upstreamResponse.StatusCode;
+                kestrelOptions.Limits.KeepAliveTimeout = keepAlive;
+            });
 
-            Log($"StatusCode: {upstreamResponse.StatusCode}");
+            builder.Services.AddControllers();
```
Comment (mikeharder) [inline tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs:45]: Do we still need this if we don't have any controllers?  It's probably a no-op so doesn't matter, but curious about best practices.

**Review by mikeharder — APPROVED** [review]

**Comment by mikeharder** [comment]
@lmolkova: I reviewed this again and it LGTM.

File: tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs (hunk 2 above)
Comment (lmolkova) [reply tools/http-fault-injector/Azure.Sdk.Tools.HttpFaultInjector/Program.cs:45]: good catch! no we don't need it anymore

**Pull request merged**